theano/sandbox/gpuarray/blas.py

```diff
@@ -239,6 +239,7 @@ def c_code(self, node, name, inputs, outputs, sub):
         dims[0] = PyGpuArray_DIMS(%(A)s)[0];
         dims[1] = PyGpuArray_DIMS(%(B)s)[1];
 
+        Py_XDECREF(%(out)s);
```

Comment:
make an issue to tell that this op do not reuse preallocated output. With scan, we start to do that more frequently.

Ideally, we should only allocate if no output or output wrong shape.